packages/adminrouter/extra/src/lib/cache.lua

```diff
@@ -602,24 +600,24 @@ local function periodically_refresh_cache(auth_token)
 
         -- Invoke timer business logic.
         refresh_cache(true, auth_token)
-
-        -- Register new timer.
-        local ok, err = ngx.timer.at(_CONFIG.CACHE_POLL_PERIOD, timerhandler)
-        if not ok then
-            ngx.log(ngx.ERR, "Failed to create timer: " .. err)
-        else
-            ngx.log(ngx.INFO, "Created recursive timer for cache updating.")
-        end
     end
 
-    -- Trigger initial timer, about CACHE_FIRST_POLL_DELAY seconds after
+    -- Trigger the timer, every CACHE_POLL_PERIOD seconds after
     -- Nginx startup.
     local ok, err = ngx.timer.at(_CONFIG.CACHE_FIRST_POLL_DELAY, timerhandler)
```

Comment:
Didn't we decide not to have an initial update? @jgehrcke

Reply:
@gpaul it's triggering the initial update with ngx.timer.at(_CONFIG.CACHE_FIRST_POLL_DELAY,...) but the comment is misleading, I will fix it now. Or do you mean something else with the "initial update"?

Reply:
2cents: Keeping the initial update code increases the chance that the incoming request will not have to wait for cache refresh. I.e. the initial update happens 2s after the initialization is complete, the scheduled update happens 25s after the init. During this 25s  it is much more likely that a request will be blocked while the cache is initialized than during 2s period which is the case when we keep the initial/short update. Also, rewriting the tests to match new logic would take some time extra time.